bottlecap/src/lifecycle/invocation/context.rs

```diff
@@ -397,6 +397,19 @@ impl ContextBuffer {
         }
     }
 
+    #[must_use]
+    pub fn get_cold_start_span(&mut self) -> Option<&mut Span> {
+        if let Some(context) = self
+            .buffer
+            .iter_mut()
+            .find(|context| context.cold_start_span.is_some())
+        {
+            return context.cold_start_span.as_mut();
+        }
+
+        None
+    }
```

Comment:
Not a fan of returning a field from a context, would it make sense to return instead the context and rename this to something like `get_context_with_cold_start` or something similar?

In reality, no other method here returns a field, but instead a context – it only modifies fields given a specific context key